### DISCUSSION TITLE
Save and Restore - Read-back PVs

### DISCUSSION THREAD

**Discussion by UrbanBobek** [ControlSystemStudio/phoebus#2466] [Q&A] [answered]
Hi all,

I would like to ask what is the purpose of read-back PVs in a save set and how they affect how the value of a PV is saved and restored.
From my understanding, the value from the read-back PV should be stored when creating a snapshot and then applied to the set-point PV when restoring the snapshot. But this is not the behavior I got when testing. The value of the set-point PV is set to the value of  the saved set-point PV no matter if a read-back PV is set or not.  In my tests I used Save and Restore service v4.7.0.

I know that an upgrade to the Save and Restore service is around the corner, so I hope this question is relevant.

Thank you!

**Comment by shroffk** [comment, marked as answer]
Yes, the question is relevant since I don't think this is documented anywhere.

The read back PV is just a reference and is designed to never be used to restore...
Think of situations were the readback might not be exactly the same as the set point. In these cases when taking the snapshot you also want to make a record of the setpoint and the readback which will be useful for the user to making decisions on restore later.

**Reply by UrbanBobek** [reply]
I see, thank you for your answer. 

Part of my confusion came from the fact that I am unable to see the stored readback values. If I enable the "show live readback" option I can see the live readback, but If I enable the "show stored readback" option, nothing changes on the display. Could this be a bug or am I missing something on my part?

**Reply by georgweiss** [reply]
Sorry for a late reply...
@UrbanBobek , you are right. The stored read-back values are not shown. That is a bug, I will create a PR as soon as I have a fix.

**Reply by UrbanBobek** [reply]
No problem, I am glad we caught the bug :)
